test/LoggerAbstractServiceFactoryTest.php

```diff
@@ -146,6 +147,53 @@ public function testRetrievesDatabaseServiceFromServiceManagerWhenEncounteringDb
         $this->assertAttributeSame($db, 'db', $writer);
     }
 
+    public function testRetrievesMongoDBServiceFromServiceManagerWhenEncounteringMongoDbWriter()
+    {
+        $mongoClient = $this->getMockBuilder('MongoClient')
```

Comment:
This test should check for existence of the mongodb extension prior, and mark the test skipped if it is not. I can make that change on merge.